src/transformers/utils/fx.py

```diff
@@ -1057,8 +1057,8 @@ def trace(
                 continue
             # We enforce that root must either be a PreTrainedModel or deserialized from a serialized traced model to
             # be able to use HFTracer._generate_dummy_input.
-            if isinstance(root, self.supported_archs) or type(root).__qualname__.startswith(
-                "_deserialize_graph_module"
+            if isinstance(root, self.supported_archs) or any(
+                type(root).__qualname__.startswith(x) for x in ["_deserialize_graph_module", "_CodeOnlyModule"]
             ):
```

Comment:
```suggestion
            if isinstance(root, self.supported_archs) or type(root).__qualname__.startswith(["_deserialize_graph_module", "_CodeOnlyModule"]):
```
this is natively supported!

Reply:
Thanks for the info.

It should be `tuple` of `str` instead of `list` though.